Dockerfile

```diff
@@ -14,50 +14,26 @@
 #   - Runs a non-root user.
 #   - Sets up the entrypoint and port.
 
-FROM python:3.8-slim-buster as base-image
+FROM tiangolo/uvicorn-gunicorn-fastapi:python3.8 as base-image
```

Comment:
My concern here is that this sort of duplicates how the fastapi version is managed since this image is setting the version of fastapi. Or is the idea that the pip installation of `requirements.txt` will upgrade or downgrade fastapi as necessary? Part of me doesn't like that the docker image is managing a bunch of Python dependencies so we lose the consistency of installing into a clean environment with just the packages that we compiled with pip-tools; I don't know if that's a practical concern?

Reply:
It's definitely a concern, and you're right that I'm not sure how pip affects that. I hadn't thought that part through and need to go read the upstream Dockerfile. What it gains is that it handles all the complicated setup for uvicorn and gunicorn automatically. Maybe we should talk about the trade-off at co-work?

Reply:
Yeah, let's chat about this at co-work. I'm not opposed to moving to a different base image; I just want to be sure we understand what we gain, lose, and how things work in the new system.